src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -132,6 +138,63 @@ private static bool HeadersAllowGzip(IDictionary<string, StringValues> headers,
             return false;
         }
 
+        protected async Task WaitForExecutionOnSpecificNode(TransactionOperationContext context, ClusterTopology clusterTopology, string node, long index)
+        {
+            using (var requester = ClusterRequestExecutor.CreateForSingleNode(clusterTopology.GetUrlFromTag(node), ServerStore.Server.Certificate.Certificate))
+            {
+                await requester.ExecuteAsync(new WaitForRaftIndexCommand(index), context);
+            }
+        }
+
+        protected async Task WaitForExecutionOnRelevantNodes(JsonOperationContext context, string database, ClusterTopology clusterTopology, List<string> members, long index)
+        {
+            if (members.Count == 0)
+                throw new InvalidOperationException("Cannot wait for execution when there are no nodes to execute ON.");
+
+            var executors = new List<ClusterRequestExecutor>();
+            var timeoutTask = TimeoutManager.WaitFor(TimeSpan.FromMilliseconds(10000));
+            var waitingTasks = new List<Task>
+            {
+                timeoutTask
+            };
+            var cts = CancellationTokenSource.CreateLinkedTokenSource(ServerStore.ServerShutdown);
+            try
+            {
+                foreach (var member in members)
+                {
+                    var url = member == ServerStore.NodeTag ?
+                        Server.WebUrl :
+                        clusterTopology.GetUrlFromTag(member);
+                    var requester = ClusterRequestExecutor.CreateForSingleNode(url, ServerStore.Server.Certificate.Certificate);
```

Comment:
Same thing here, actually. We are creating new instances from an instance method, which seems strange.